PORT-7715 | Make Dora Metrics Script Platform Agnostic (#1111)
port-labs/port-docs: docs/build-your-software-catalog/custom-integration/api/ci-cd/github-workflow/guides/dora-metrics.md

          pip install -r dora/requirements.txt

      - name: Compute PR Metrics
        run: |

>           python dora/calculate_pr_metrics.py  --owner ${{ matrix.owner }} --repo ${{ matrix.repository }} --token ${{ secrets.PATTOKEN }} --timeframe ${{env.TIMEFRAME_IN_DAYS}} --platform github-actions

Reviewer: in compute PR metrics run script I think we need --ignore_workflows

Author: It was deliberately left out in the github action to enforce running with workflows by default

Author: to run the without computing workflow, we can set that parameter

```python showLineNumbers title="deployment_frequency.py"
import datetime
import os
import json
from github import Github
import argparse
import logging

#Throttling, set to None to restore default behavior

> SECONDS_BETWEEN_REQUESTS=0.12

Reviewer 2: Are these throttling parameters needed?

    parser.add_argument('--token', required=True, help='GitHub token')
    parser.add_argument('--workflows', required=True, help='GitHub workflows as a JSON string.')
    parser.add_argument('--branch', default='main', help='Branch name')
    parser.add_argument('--timeframe', type=int, default=30, help='Timeframe in days')
    parser.add_argument('--platform', default='github-actions', choices=['github-actions', 'self-hosted'], help='CI/CD platform type')
    args = parser.parse_args()

    deployment_frequency = DeploymentFrequency(args.owner, args.repo, args.workflows, args.branch, args.timeframe, pat_token = args.token)
    report = deployment_frequency()
    print(report)
    
    if args.platform == "github-actions":
       with open(os.getenv("GITHUB_ENV"), "a") as github_env:
           github_env.write(f"deployment_frequency_report={report}\n")
```
</details>

<details>
  <summary><b>Lead Time For Changes</b></summary>

```python showLineNumbers title="lead_time_for_changes.py"
import datetime
import os
import json
from github import Github
import argparse
import logging

#Throttling, set to None to restore default behavior

> SECONDS_BETWEEN_REQUESTS=0.12

Reviewer: what is the significance of this 0.12 and I didn't get the throttling comment here ?

> SECONDS_BETWEEN_REQUESTS=0.12

Reviewer 2: Are these throttling parameters needed?

Author: Not neccessarily, but the default values are a bit high so i overwrote them with lower values

Author: https://pygithub.readthedocs.io/en/stable/changes.html#version-2-1-0-september-29-2023:~:text=%C2%B6-,Request%20throttling,-This%20release%20introduces

    parser.add_argument('--repo', required=True, help='Repository name')
    parser.add_argument('--token', required=True, help='GitHub token')
    parser.add_argument('--workflows', default='[]', help='GitHub workflows as a JSON string.')
    parser.add_argument('--branch', default='main', help='Branch name')
    parser.add_argument('--timeframe', type=int, default=30, help='Timeframe in days')
    parser.add_argument('--platform', default='github-actions', choices=['github-actions', 'self-hosted'], help='CI/CD platform type')
    parser.add_argument('--ignore_workflows', action='store_true', help='Exclude workflows. Default is False.')
    args = parser.parse_args()

    lead_time_for_changes = LeadTimeForChanges(
        args.owner, args.repo, args.workflows, args.branch, args.timeframe, pat_token=args.token,ignore_workflows=args.ignore_workflows
    )
    report = lead_time_for_changes()
    print(report)
    
    if args.platform == "github-actions":
       with open(os.getenv("GITHUB_ENV"), "a") as github_env:
           github_env.write(f"lead_time_for_changes_report={report}\n")
```
</details>

> Congrats 🎉 You've successfully scheduled a GitHub action to periodically ingest estimated `DORA Metrics` for GitHub repository(s).

Reviewer: this note can be changes as now you have schedule a job to periodically ingest estimated DORA metics for git repos.

> Congrats 🎉 You've successfully scheduled a GitHub action to periodically ingest estimated `DORA Metrics` for GitHub repository(s).

Reviewer 2: ```suggestion
Congrats 🎉 You've successfully created a scheduled GitHub workflow that periodically calculates and ingests `DORA Metrics` for GitHub repository(s).

```
